exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -568,182 +883,362 @@ func (e *Exchange) CancelBatchOrders(ctx context.Context, o []order.Cancel) (*or
 		default:
 			return nil, order.ErrOrderIDNotSet
 		}
-	}
-	cancelledOrders, err := e.CancelMultipleOrdersByIDs(ctx, orderIDs, clientOrderIDs)
-	if err != nil {
-		return nil, err
+		if assetType == asset.Futures {
+			if o[i].Pair.IsEmpty() {
+				return nil, currency.ErrSymbolStringEmpty
+			} else if o[0].Pair != o[i].Pair { //nolint:gosec // index 0 safe because loop requires len(o) > 0
+				return nil, currency.ErrPairNotFound
+			}
+		}
 	}
 	resp := &order.CancelBatchResponse{
 		Status: make(map[string]string),
 	}
```

Comment:
🚧 Change request:
I'm not happy with how we're shifting responsibility here.
I think we should:
* Include all failures in a returned `errs`
* Include Cancelled bool, Error, State and Message in the resp status struct
* Use the original order itself as the map key, instead of creating a struct of (potentially clashing) orderids and clientIds
🌴 This pattern applies to all assets here.

Reply:
@samuael You gave a thumbs up but didn't action.